src/include/hardware.h

```diff
@@ -25,6 +25,23 @@
         WT_READ_BARRIER();      \
     } while (0)
 
+/*
+ * In some architectures with weak memory ordering, the CPU can reorder the reads across full
+ * barriers in other threads. Guarantee that subsequent reads do not see any earlier state in those
+ * architectures.
+ */
+#define WT_ORDERED_READ_WEAK_MEMORDER(v, val) \
```

Comment:
I don't understand the need for `WT_ORDERED_READ_WEAK_MEMORDER`, this should be the implementation of `WT_ORDERED_READ`, we don't need two of them. Are you worried that changing `WT_ORDERED_READ` would be too risky?

Reply:
Yes - this is a temporary risk-mitigation strategy. We found that adding read barriers into the search path of the skip list caused a noticeable performance regression on x86, and are looking for a low-risk mechanism for avoiding that regression.

Reply:
I suggest adding another paragraph to this comment, telling developers not to use this macro, and explaining that it's a temporary solution to a performance regression. It's a bit awkward to link a Jira ticket to make the permanent change - since that is currently a project and not a single public ticket.

Reply:
done

Reply:
I still think introducing `WT_ORDERED_READ_WEAK_MEMORDER` and `WT_READ_ONCE` is a bit risky in itself, since it's not clear when to use them compared to other primitives, but I understand the desire to only change the specific functions that created the regression.